File: chp3/python/Soy.py
from CondimentDecorator import CondimentDecorator


class Soy(CondimentDecorator):
    def __init__(self, beverage):
        super(Soy, self).__init__()
        self.beverage = beverage

    def getDescription(self):
        return self.beverage.getDescription() + ", Soy"

    def cost(self):
        return self.beverage.cost() + 0.15

File: chp3/java/src/Soy.java
public class Soy extends CondimentDecorator {
    public Beverage beverage;

    public Soy(Beverage beverage) {
	this.beverage = beverage;
    }

    public String getDescription() {
	return beverage.getDescription() + ", Soy";
    }

    public double cost() {
	return beverage.cost() + .15;
    }
}
